tests/unittests/caffe2ImporterTest.cpp

```diff
@@ -53,3 +53,209 @@ TEST(caffe2, importConv) {
+  Tensor *inputs[] = {&inputs_0, &inputs_1, &inputs_2};
+  for (size_t i = 0; i < 3; ++i) {
+    const auto inputsHandle = inputs[i]->getHandle();
+    EXPECT_TRUE(llvm::isa<Variable>(concatNode->getNthInput(i)));
+    EXPECT_TRUE(llvm::cast<Variable>(concatNode->getNthInput(i))
+                    ->getPayload()
+                    .isEqual(*inputs[i]));
+
+    for (size_t row = 0; row < 10; ++row) {
+      for (size_t column = 0; column < 7; ++column) {
+        EXPECT_FLOAT_EQ(result.at({row, i * 7 + column}),
+                        inputsHandle.at({row, column}));
+      }
+    }
+  }
+}
+
+/// Test loading a batched matmul with transpose on RHS.
+TEST(caffe2, batchedMatmulRHS) {
+  ExecutionEngine EE{BackendKind::Interpreter};
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+   std::string NetDescFilename(
+      "tests/models/caffe2Models/matmul_trans_RHS_predict_net.pbtxt");
+  std::string NetWeightFilename(
+      "tests/models/caffe2Models/empty_init_net.pbtxt");
+   SaveNode *output;
+  Tensor inputs_0(ElemKind::FloatTy, {3, 10, 7});
+  Tensor inputs_1(ElemKind::FloatTy, {10, 7});
+  inputs_0.getHandle().randomize(-3.0, 3.0, mod.getPRNG());
+  inputs_1.getHandle().randomize(-3.0, 3.0, mod.getPRNG());
+  // Destroy the loader after the graph is loaded since the following execution
+  // will not depend on anyting from the loader.
+  {
+    caffe2ModelLoader caffe2LD(NetDescFilename, NetWeightFilename,
+                               {"inputs_0", "inputs_1"}, {&inputs_0, &inputs_1},
+                               *F);
+    output = caffe2LD.getSingleOutput();
+  }
+   auto result = output->getVariable()->getHandle();
+  // Check that the shape of the output matches what Caffe2 expects.
+  std::vector<size_t> expectedDims = {3, 10, 10};
+  EXPECT_TRUE(result.dims().vec() == expectedDims);
+   // High level check on the content of the graph.
+  // We have 1 transpose, 1 matmul, 1 save, and 2 reshapes.
+  EXPECT_EQ(F->getNodes().size(), 5);
+  // With have 2 inputs and one outputs.
+  EXPECT_EQ(mod.getVars().size(), 3);
+   // Check that the graph has the expected shape,
+  // starting from the output.
+   // Batched matmul with broadcasted RHS are lowered
+  // to a regular matmul, where LHS is reshaped from
+  // a 3D tensor to a flattened matrix.
+  Node *reshapeResult = output->getNthInput(0).getNode();
+  EXPECT_TRUE(llvm::isa<ReshapeNode>(reshapeResult));
+   Node *matmul = reshapeResult->getNthInput(0).getNode();
+  EXPECT_TRUE(llvm::isa<MatMulNode>(matmul));
+  const size_t matmulDims[] = {30, 10};
+  EXPECT_EQ(matmul->dims(0), llvm::makeArrayRef(matmulDims));
+   NodeValue lhs = matmul->getNthInput(0);
+  EXPECT_TRUE(llvm::isa<ReshapeNode>(lhs.getNode()));
+  Node *lhsInput = lhs.getNode()->getNthInput(0).getNode();
+  EXPECT_TRUE(llvm::isa<Variable>(lhsInput));
+  EXPECT_TRUE(llvm::cast<Variable>(lhsInput)->getPayload().isEqual(inputs_0));
+   NodeValue rhs = matmul->getNthInput(1);
```

Comment:
Is it feasible to use `getRHS()` instead of indexed input? Here and everywhere else?

Reply:
Done